smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatIntegrationTest.java

```diff
@@ -113,23 +156,514 @@ public void processMessage(Message message) {
         });
 
         MucCreateConfigFormHandle handle = mucAsSeenByOne.createOrJoin(Resourcepart.from("one-" + randomString));
-        if (handle != null) {
-            handle.makeInstant();
+        try {
+            if (handle != null) {
+                handle.makeInstant();
```

Comment:
I know this pattern was used before, but since this PR repeats this pattern I wonder if it is actually a good idea. All (or maybe just most?) MUCs used by the integration test run should be freshly created, shouldn't they? So why do we use `MultiUserChat.createOrJoin()` when you could simply use `MulitUserChat.create()` and get rid of the `if (handle != null)`?

If you agree and are motivated to give it a shot, then please do. Otherwise we could look into it after your PR has been merged.

Reply:
Tried this. Broke plenty. I've re-added it, but abstracted it into a method to make it read cleaner.